data-types/timedelta64/README.md

```diff
@@ -0,0 +1,95 @@
+
+While it is possible to construct a NumPy `timedelta64` data type with a scaling factor of `0`, NumPy will automatically normalize this to `1`.
+
+#### Unit
+The NumPy `timedelta64` data type takes a unit parameter, which must be one of the following temporal units:
+
+| Identifier | Meaning     |
+|------------|----------|
+| Y        | year   |
+| M        | month   |
+| W       | week     |
+| D        | day      |
+| h       | hour     |
+| m      | minute    |
+| s       | second     |
+| ms       | millisecond     |
+| us       | microsecond     |
+| μs       | microsecond     |
+| ns       | nanosecond      |
+| ps       | picosecond      |
+| fs       | femtosecond     |
+| as       | attosecond     |
+
+> Note: "us" and "μs" are treated as equivalent by NumPy.
+
+> Note: NumPy permits the creation of `timedelta64` data types with an unspecified unit. In this case, the unit is set to the special value `"generic"`.
+
+#### Endianness
+The NumPy `timedelta64` data type takes a byte order parameter, which must be either little-endian or big-endian. 
+
+## Data type representation
+
+### Name
+
+The name of this data type is the string `"timedelta64"`.
+
+### Configuration
+
+This data type requires a configuration. The configuration for this data type is a JSON object with the following fields:
+
+| field name | type | required | notes |
+|------------|----------|---|---|
+| `"unit"` | one of: `"Y"`, `"M"` , `"W"`, `"D"` , `"h"` , `"m"` , `"s"` , `"ms"` , `"us"` , `"μs"` , `"ns"` , `"ps"` , `"fs"` , `"as"`, `"generic"` | yes | None |
+| `"scale_factor"` | `integer` | yes | The number must represent an integer from the inclusive range `[1, 2147483647]` |
+
+> Note: the NumPy `timedelta64` data type is parametrized by an endianness (little or big), but the Zarr `timedelta64` data type is not. In Zarr, the endianness of `timedelta64` arrays is determined by the configuration of the `codecs` metadata and is thus not part of the data type configuration.
+
+> Note: as per NumPy, `"us"` and `"μs"` are equivalent and interchangeable representations of microseconds.
+
+No additional fields are permitted in the configuration.
+
+### Examples
+The following is an example of the metadata for a `timedelta64` data type with a unit of microseconds and a scale factor of 10. This configuration defines a data type equivalent to the NumPy data type `timedelta64[10us]`:
+
+```json
+{
+    "name": "timedelta64",
+    "configuration": {
+        "unit": "us",
+        "scale_factor": 10
+    }
+}
+```
+
+## Fill value representation
+
+`timedelta64` fill values are represented as one of:
+- a JSON number with no fraction or exponent part that is within the range `[-2^63, 2^63 - 1]`. 
+- the string `"NaT"`, which denotes the value `NaT`. 
+
+> Note: the `NaT` value may optionally be encoded as the JSON number `-9223372036854775808`, i.e., `-2^63`. 
```

Comment:
I don't understand this part. Here it seems like the use can configure their own custom fill value? If so, shouldn't `fill_value` be in `configuration`. And what would be the use case for that?

Isn't it simpler if we just say that, like numpy, the integer `-2^63` represents NaT.

Reply:
what I'm trying to say here is that the following two cases are the only acceptable fill values:

```json
"fill_value" : <a JSON integer in the range [-2^63, 2^63]>
```

```json
"fill_value" : "NaT"
```

With one degenerate case:

```json
"fill_value": "NaT"
```
has the same meaning as 
```json
"fill_value": -9223372036854775808
```

Reply:
the statement "`timedelta64` fill values are represented as one of" is intended to mean "there are two possible forms for the `"fill_value"` metadata. Maybe I should make this clear. I definitely don't want to convey that users can configure a custom fill value.